### PR TITLE
color and names

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: lib/listTasks.js [hunk 1]
```diff
@@ -2,15 +2,33 @@ module.exports = (data, title, filter) => {
...
-      fields: [
-        { title: 'Slot', value: d.Slot, short: true },
-        { title: 'NodeID', value: d.NodeID, short: true },
-        { title: 'Status', value: d.Status.State, short: true },
-        { title: 'Since', value: d.Status.Timestamp, short: true }
-      ]
-    }))
+    attachments: data.sort((a, b) => a.Slot - b.Slot).map((d) => {
+      const attachment = {
+        title: `Task ID: ${d.ID}`,
```
Comment (jgallen23) [inline lib/listTasks.js]: this should be name instead of id

Comment (jgallen23) [reply lib/listTasks.js]: move task id into fields

File: lib/listTasks.js [hunk 2]
```diff
@@ -2,15 +2,33 @@ module.exports = (data, title, filter) => {
...
+        fields: [
+          { title: 'Slot', value: d.Slot, short: true },
+          { title: 'ServiceID', value: d.ServiceID, short: true },
+          { title: 'Status', value: d.Status.State, short: true },
+          { title: 'Since', value: d.Status.Timestamp, short: true }
```
Comment (jgallen23) [inline lib/listTasks.js]: lets add image here too

File: lib/listTasks.js [hunk 3]
```diff
@@ -2,15 +2,33 @@ module.exports = (data, title, filter) => {
...
-        { title: 'Status', value: d.Status.State, short: true },
-        { title: 'Since', value: d.Status.Timestamp, short: true }
-      ]
-    }))
+    attachments: data.sort((a, b) => a.Slot - b.Slot).map((d) => {
+      const attachment = {
+        title: `Task ID: ${d.ID}`,
+        fields: [
+          { title: 'Slot', value: d.Slot, short: true },
+          { title: 'ServiceID', value: d.ServiceID, short: true },
```
Comment (jgallen23) [inline lib/listTasks.js]: is there a service name too or just id?

File: lib/listTasks.js [hunk 4]
```diff
@@ -2,15 +2,33 @@ module.exports = (data, title, filter) => {
...
-    attachments: data.sort((a, b) => a.Slot - b.Slot).map((d) => ({
-      title: `Task ID: ${d.ID}`,
-      fields: [
-        { title: 'Slot', value: d.Slot, short: true },
-        { title: 'NodeID', value: d.NodeID, short: true },
-        { title: 'Status', value: d.Status.State, short: true },
-        { title: 'Since', value: d.Status.Timestamp, short: true }
-      ]
-    }))
+    attachments: data.sort((a, b) => a.Slot - b.Slot).map((d) => {
```
Comment (jgallen23) [inline lib/listTasks.js]: lets actually sort by State instead

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: lib/listTasks.js (hunk 3 above)
Comment (orthagonal) [reply lib/listTasks.js]: Don't see a name in there.

File: lib/listTasks.js [hunk 5]
```diff
@@ -2,15 +2,43 @@ module.exports = (data, title, filter) => {
...
+          { title: 'Status', value: d.Status.State, short: true },
+          { title: 'Since', value: d.Status.Timestamp, short: true }
+        ]
+      };
+      // add container id if is available:
+      if (d.Status && d.Status.ContainerStatus && d.Status.ContainerStatus.ContainerID) {
+        attachment.fields.splice(2, 0, { title: 'Container', value: d.Status.ContainerStatus.ContainerID, short: true });
+      }
+      // add container image if available:
+      if (d.Spec && d.Spec.ContainerSpec && d.Spec.ContainerSpec.Image) {
```
Comment (jgallen23) [inline lib/listTasks.js:33]: how would it not be available?

File: lib/listTasks.js [hunk 6]
```diff
@@ -2,15 +2,43 @@ module.exports = (data, title, filter) => {
...
+      return p.indexOf(a.Status.State) - p.indexOf(b.Status.State);
+    }).map((d) => {
+      const attachment = {
+        title: `Task: ${d.Name}`,
+        fields: [
+          { title: 'Task Id', value: d.ID, short: true },
+          { title: 'Slot', value: d.Slot, short: true },
+          { title: 'ServiceID', value: d.ServiceID, short: true },
+          { title: 'Status', value: d.Status.State, short: true },
+          { title: 'Since', value: d.Status.Timestamp, short: true }
```
Comment (jgallen23) [inline lib/listTasks.js]: can we do relative time here?

**Pull request merged**